src/Data/Foldable.purs

```diff
@@ -449,10 +447,13 @@ minimum = minimumBy compare
 -- | function. The comparison function should represent a total ordering (see
 -- | the `Ord` type class laws); if it does not, the behaviour is undefined.
 minimumBy :: forall a f. Foldable f => (a -> a -> Ordering) -> f a -> Maybe a
-minimumBy cmp = foldl min' Nothing
+minimumBy cmp = foldlMaybe \x y -> if cmp x y == LT then x else y
+
+foldlMaybe :: forall f a . Foldable f => (a -> a -> a) -> f a -> Maybe a
+foldlMaybe f = foldl f' Nothing
```

Comment:
In many ways, this feel like `foldl1`...

Reply:
@JordanMartinez You are right, they are the same but the difference is that `foldlMaybe` works on zero length. I was not sure how should I name the function so I looked to Hoogle and found [foldl1May](https://hackage.haskell.org/package/safe-0.3.19/docs/Safe-Foldable.html#v:foldl1May) however I didn't see the `1` in the name so I chosed `foldlMaybe`. Now I see `foldl1` is a better name.

Reply:
However if I name it `foldl1` it make break code if someone open imported Traverse and foldl1 from Traverse1 in the same file.

Reply:
I think `foldlMaybe` makes more sense than `fold1Maybe` or `fold1May`. The `1` indicates the container will be non-empty. Since a `Foldable` container can be empty, dropping the `1` makes sense to me.